src/Log/Engine/BaseLog.php

```diff
@@ -90,6 +91,46 @@ public function scopes()
      */
     protected function _format(string $message, array $context = []): string
     {
-        return $message;
+        if (strpos($message, '{') === false && strpos($message, '}') === false) {
+            return $message;
+        }
+
+        preg_match_all('/\{([a-z][a-z0-9-_]*)\}/i', $message, $matches);
```

Comment:
Do you want to use `Text::insert()`? It would require adding cakephp/utility as a dependency, and pre-casting data into strings.

Reply:
> It would require adding cakephp/utility as a dependency

That's the reason I didn't use it. The code of Text::insert() looks lot more expensive than this regex check.